// @keystone-next/app-version-legacy/dist/app-version-legacy.cjs.prod.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var _defineProperty = require('@babel/runtime/helpers/defineProperty');
var accessControlLegacy = require('@keystone-next/access-control-legacy');

class AppVersionProvider {
  constructor({
    version,
    access,
    schemaNames
  }) {
    _defineProperty(this, "_access", void 0);

    _defineProperty(this, "_version", void 0);

    this._access = accessControlLegacy.parseCustomAccess({
      access: access,
      schemaNames,
      defaultAccess: true
    });
    this._version = version;
  }

  getTypes({}) {
    return [];
  }

  getQueries({
    schemaName
  }) {
    return this._access[schemaName] ? [`"""The version of the Keystone application serving this API."""
          appVersion: String`] : [];
  }

  getMutations({}) {
    return [];
  }

  getSubscriptions({}) {
    return [];
  }

  getTypeResolvers({}) {
    return {};
  }

  getQueryResolvers({
    schemaName
  }) {
    return this._access[schemaName] ? {
      appVersion: () => this._version
    } : {};
  }

  getMutationResolvers({}) {
    return {};
  }

  getSubscriptionResolvers({}) {
    return {};
  }

}
function appVersionMiddleware(version) {
  return (req, res, next) => {
    res.set('X-Keystone-App-Version', version);
    next();
  };
}

exports.AppVersionProvider = AppVersionProvider;
exports.appVersionMiddleware = appVersionMiddleware;
